odoo/orm/models.py

```diff
@@ -4174,7 +4174,7 @@ def write(self, vals: ValuesType) -> typing.Literal[True]:
             for fields in determine_inverses.values():
                 # write again on non-stored fields that have been invalidated from cache
                 for field in fields:
-                    if not field.store and any(field._cache_missing_ids(real_recs)):
+                    if not field.store and (not field.inherited or field.type not in ["one2many", "many2many"]) and any(field._cache_missing_ids(real_recs)):
```

Comment:
Makes sense.  Just a minor comment: tuples are faster than lists (in the ORM, we care about such details).
```suggestion
                    if not field.store and (not field.inherited or field.type not in ('one2many', 'many2many')) and any(field._cache_missing_ids(real_recs)):
```
Can you point out the use-case that makes this change necessary?
We need to add a test case for it in module `test_orm`.

Reply:
I was surprised by this statement, so here is my benchmark :eyes: 

```python
Python 3.11.12 (main, Apr  9 2025, 08:55:54) [GCC 11.4.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.22.2 -- An enhanced Interactive Python. Type '?' for help.

In [1]: a = 'hibou'

In [2]: LIST = ['coucou', 'hibou']

In [3]: TUPLE = ('coucou', 'hibou')

In [4]: %timeit a not in TUPLE
30.6 ns ± 0.937 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [5]: %timeit a not in LIST
30.7 ns ± 1.62 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [6]: %timeit a not in ('coucou', 'hibou')
31.1 ns ± 0.432 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [7]: %timeit a not in ['coucou', 'hibou']
29.6 ns ± 1.35 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```